src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2144,7 +2341,17 @@ mgp_error mgp_graph_detach_delete_vertex(struct mgp_graph *graph, mgp_vertex *ve
     if (!MgpGraphIsMutable(*graph)) {
       throw ImmutableObjectException{"Cannot remove a vertex from an immutable graph!"};
     }
-    const auto result = graph->impl->DetachRemoveVertex(&vertex->impl);
+    const auto result = std::visit(
+        memgraph::utils::Overloaded{
+            [vertex](memgraph::query::DbAccessor *impl) {
```

Comment:
Use one lambda with auto*. Remove Overloaded

Reply:
I think I can't use auto* since in one case I am sending VertexAccessor and in other SubgraphVertexAccessor. And also I am doing MG_ASSERT on correct type for SubgraphDbAccessor so that inside that function I have SubgraphVertexAccessor.

Reply:
I think this too is resolved.